orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactUtils.java

```diff
@@ -139,7 +141,30 @@ private List<Artifact> getAllArtifacts(
         contextParameterProcessor.process(
             boundArtifactMap, contextParameterProcessor.buildExecutionContext(stage), true);
 
-    return objectMapper.convertValue(evaluatedBoundArtifactMap, Artifact.class);
+    Artifact evaluatedArtifact =
+        objectMapper.convertValue(evaluatedBoundArtifactMap, Artifact.class);
+    return getBoundInlineArtifact(evaluatedArtifact, stage.getExecution())
+        .orElse(evaluatedArtifact);
+  }
+
+  private Optional<Artifact> getBoundInlineArtifact(
+      @Nullable Artifact artifact, PipelineExecution execution) {
+    if (ObjectUtils.anyNull(
+        artifact, execution.getTrigger(), execution.getTrigger().getArtifacts())) {
+      return Optional.empty();
+    }
+    try {
+      ExpectedArtifact expectedArtifact =
+          ExpectedArtifact.builder().matchArtifact(artifact).build();
+      return ArtifactResolver.getInstance(execution.getTrigger().getArtifacts(), true)
+          .resolveExpectedArtifacts(List.of(expectedArtifact))
+          .getResolvedArtifacts()
+          .stream()
+          .findFirst();
+    } catch (InvalidRequestException e) {
```

Comment:
What is throwing InvalidRequestException that we might catch here?

Reply:
[`ArtifactResolver.resolveExpectedArtifacts()`](https://github.com/spinnaker/orca/blob/a4a82084a48cfc323e3b0b04b670eb17125a3ece/orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/ArtifactResolver.java#L127) throws this if it's unable to resolve the artifact. It's not something we care about, as most often it will probably not be matched.